index.html

```diff
@@ -7,7 +7,35 @@
     <title>Vite App</title>
   </head>
   <body>
-    <div id="app"></div>
+    <div id="app">
+        <button id="tooltip_on_hover">
+            Click me!
+            <span class="tooltip">
+                Hello JS World!
+            </span>
+        </button>
+        <button id="tooltip_on_click"> Click me!
+            <span class="tooltip">
+                Hello JS World!
+            </span>
+        </button>
+        <button id="modal_on_click" data-toggle="modal" data-target="#my_modal"> Click me! </button>
```

Comment:
Great idea to use data-attributes to define behaviour, unfortunate that you didn't use them :D

You can read here about right using them:
https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes